website/docs/r/edgegateway.html.markdown

```diff
@@ -0,0 +1,50 @@
+---
+layout: "vcd"
+page_title: "vCloudDirector: vcd_edgegateway"
+sidebar_current: "docs-vcd-resource-edgegateway"
+description: |-
+  Provides a vCloud Director edge gateway. This can be used to create and delete edge gateways connected to one or more external networks.
+---
+
+# vcd\_edgegateway
+
+Provides a vCloud Director edge gateway directly connected to one or more external networks. This can be used to create
+and delete edge gateways for Org VDC networks to connect.
+
+Supported in provider *v2.4+*
+
+~> **Note:** Only `System Administrator` can create an edge gateway.
+You must use `System Adminstrator` account in `provider` configuration
+and then provide `org` and `vdc` arguments for edge gateway to work.
+
+## Example Usage
+
+```hcl
```

Comment:
maybe more sophisticated example with external network creation and linking to it? cc: @lvirbalas

Reply:
I believe the example in the docs should be as simple as possible. We will have more sophisticated examples in the templates.

Reply:
well my note from experience of my PR and Linas feedback :))

Reply:
If needed, I can put something similar to [the "complex" test in the code](https://github.com/dataclouder/terraform-provider-vcd/blob/81686371745f36392bf538d71745d76881984e8e/vcd/resource_vcd_edgegateway_test.go#L159)

Reply:
I agree to Vaidotas' comment here. We should follow more prescriptive example model, because we saw that "dry" examples are not enough for the user to understand how to use the system. On the other hand, I do agree that a too long example is not good either. So, we should really search for the golden balance of examples which are holistic, but not overwhelming...

Reply:
An example with an external network as suggested will triple the lines without adding much.
Shall I add instead a routed network that uses the new edge gateway?

Reply:
### Candidate example 1
```hcl
resource "vcd_external_network" "extnet2" {
  name        = "extnet2"
  description = "Second External Network"

  vsphere_network {
    vcenter = "vc1"
    name    = "TestbedPG"
    type    = "DV_PORTGROUP"
  }

  ip_scope {
    gateway      = "192.168.30.49"
    netmask      = "255.255.255.240"
    dns1         = "192.168.0.164"
    dns2         = "192.168.0.196"
    dns_suffix   = "company.biz"

    static_ip_pool {
      start_address = "192.168.30.51"
      end_address   = "192.168.30.62"
    }
  }

  retain_net_info_across_deployments = "false"
}

resource "vcd_edgegateway" "TestEdgeGatewayComplex" {
  org                   = "my-org"
  vdc                   = "my-vdc"
  name                  = "test_edge_gateway_complex"
  description           = "My edge gateway with two external networks"
  gateway_configuration = "compact"
  default_gateway       = "${vcd_external_network.extnet2.name}"
  advanced              = true
  external_networks     = [ "extnet1", "${vcd_external_network.extnet2.name}" ]
}
```
### Candidate example 2
```hcl
resource "vcd_edgegateway" "egw" {
  org = "my-org"
  vdc = "my-vdc"

  name                    = "my-egw"
  description             = "new edge gateway"
  gateway_configuration   = "compact"
  default_gateway_network = "my-ext-net1"
  external_networks       = [ "my-ext-net1", "my-ext-net2" ]
}

resource "vcd_network_routed" "rnet1" {
  name         = "rnet1"
  org          = "my-org"
  vdc          = "my-vdc"
  edge_gateway = "${vcd_edgegateway.egw.name}"
  gateway      = "192.168.2.1"

  static_ip_pool {
    start_address = "192.168.2.2"
    end_address   = "192.168.2.100"
  }
}
```

Reply:
Example 2 is nice and useful!

Reply:
Added example